compiler/rustc_lint/messages.ftl

```diff
@@ -415,6 +415,16 @@ lint_improper_ctypes_union_layout_help = consider adding a `#[repr(C)]` or `#[re
 lint_improper_ctypes_union_layout_reason = this union has unspecified layout
 lint_improper_ctypes_union_non_exhaustive = this union is non-exhaustive
 
+lint_incompatible_target_modifiers =
+    mixing `{$flag_name_suffixed}` will cause an ABI mismatch
+    .note1 = `{$flag_name_suffixed}`=`{$flag_local_value}` in crate `{$local_crate}`, `{$flag_name_suffixed}`=`{$flag_extern_value}` in crate `{$extern_crate}`
+    .help = This error occurs because the `{$flag_name_suffixed}` flag modifies the ABI,
```

Comment:
Ideally this should say something like:

```suggestion
    .help = `{$flag_name_suffixed}` modifies the ABI and Rust crates compiled with different values of this flag cannot be used together safely
```

It would be good if we could provide the whole option, not just the name, so `-Zregparm` rather than `regparm`.

Reply:
Ok, I will do, but this help text from RFC: https://github.com/rust-lang/rfcs/pull/3716, maybe add comment to RFC pull request too?
```
error: mixing -Zreg-struct-return will cause an ABI mismatch

help: This error occurs because the -Zreg-struct-return flag modifies the ABI, 
      and different crates in your project were compiled with inconsistent
      settings.
help: To resolve this, ensure that -Zreg-struct-return is set to the same value
      for all crates during compilation.
help: To ignore this error, recompile with the following flag:
      -Cunsafe-allow-abi-mismatch=reg-struct-return
```

Reply:
I think something along the lines of these two..

```
help: This error occurs because the -Zreg-struct-return flag modifies the ABI, 
      and different crates in your project were compiled with inconsistent
      settings.
help: To resolve this, ensure that -Zreg-struct-return is set to the same value
      for all crates during compilation.
```

..could be useful to incorporate, we already cover some of it

Reply:
Sorry, I didn't get it. Could you, pls, rephrase it someway? What should I do, combine those two `help` message lines into one?